File: src/RestaurantPos.Client.Flutter/lib/presentation/blocs/pos_terminal/input_buffer_state.dart
import 'package:equatable/equatable.dart';

enum InputBufferTarget { none, quantity, price, comment }

class InputBufferState extends Equatable {
  final InputBufferTarget activeField;
  final String currentBuffer;

  const InputBufferState({
    required this.activeField,
    required this.currentBuffer,
  });

  const InputBufferState.empty()
      : activeField = InputBufferTarget.none,
        currentBuffer = '';

  @override
  List<Object?> get props => [activeField, currentBuffer];
}

File: src/RestaurantPos.Client.Maui/ViewModels/InputBufferState.cs
namespace RestaurantPos.Client.Maui.ViewModels;

public enum InputBufferTarget
{
    None = 0,
    Quantity = 1,
    CustomPrice = 2,
    Barcode = 3
}

public class InputBufferState
{
    public string CurrentBuffer { get; set; } = string.Empty;
    public InputBufferTarget ActiveField { get; set; } = InputBufferTarget.None;

    public void Clear()
    {
        CurrentBuffer = string.Empty;
    }

    public void AppendDigit(int digit)
    {
        CurrentBuffer += digit.ToString();
    }

    public void Backspace()
    {
        if (CurrentBuffer.Length > 0)
        {
            CurrentBuffer = CurrentBuffer.Substring(0, CurrentBuffer.Length - 1);
        }
    }
}
